subprojects/core/src/main/java/org/gradle/internal/jvm/UnsupportedJavaRuntimeException.java

```diff
@@ -18,15 +18,26 @@
 
 import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
+import org.gradle.util.DeprecationLogger;
 import org.gradle.util.GradleVersion;
 
 public class UnsupportedJavaRuntimeException extends GradleException {
+    public static final String JAVA7_DEPRECATION_WARNING = "Support for running Gradle using Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0. "
+        + "Please see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_cross_compilation for more details.";
```

Comment:
Should we use `DocumentationRegistry` to get the right URL to the userguide?

Reply:
Yes, you're right. I will do that.

Reply:
Good catch!